test/provider/database.unittest.ts

```diff
@@ -0,0 +1,210 @@
+import 'jest';
+import 'fake-indexeddb/auto';
+
+import {Database, CollectionMap} from '../../src/provider/model/database/Database';
+import {StoredNotification} from '../../src/provider/model/StoredNotification';
+import {NotificationInternal} from '../../src/client/internal';
+import {Collection} from '../../src/provider/model/database/Collection';
+
+let database: Database;
+let collection: Collection<StoredNotification>;
+let generatedNotificationCount: number = 0;
+
+beforeAll(async () => {
+    database = await new Database().delayedInit();
+    collection = database.get(CollectionMap.NOTIFICATIONS);
+});
+
+describe('Database Methods', () => {
+    describe('Get', () => {
+        it('Returns a collection when given a valid name', async () => {
+            const localCollection: Collection<StoredNotification> = database.get(CollectionMap.NOTIFICATIONS);
+            expect(localCollection).toBeDefined();
+        });
+
+        it('Throws an error when given an invalid name', async () => {
+            expect(() => {
+                database.get('INVALID COLLECTION' as any);
+            }).toThrowError();
+        });
+    });
+});
+
+describe('Collection Methods', () => {
+    afterEach(async () => {
+        const records = await collection.getAll();
+        await Promise.all(records.map(record => collection.delete(record.id)));
+
+        if ((await collection.getAll()).length !== 0) {
+            throw new Error('Collection cleanup failed: Collection not cleared!');
+        }
+    });
+
+    describe('Get', () => {
+        it('Returns a single record when passed a valid ID', async () => {
+            const note = generateNotification();
+            await collection.upsert(note);
+
+            const record = await collection.get(note.id);
+            expect(record).toEqual(note);
+        });
+
+        it('Returns undefined when passed an invalid ID', async () => {
+            const note = generateNotification();
+            await collection.upsert(note);
+
+            const record = await collection.get('INVALID ID');
+            expect(record).toBeUndefined();
+        });
+    });
+
+    describe('GetAll', () => {
+        it('Returns all records from collection', async () => {
+            const notes = [generateNotification(), generateNotification(), generateNotification()];
+            await collection.upsert(notes);
+
+            const results = await collection.getAll();
+            expect(results).toEqual(notes);
+        });
+
+        it('Returns an empty array when no records are in the collection', async () => {
+            const results = await collection.getAll();
+            expect(results).toEqual([]);
+        });
+    });
+
+    describe('GetMany', () => {
+        it('Returns an array of all records for valid IDs passed in', async () => {
+            const notes = [generateNotification(), generateNotification(), generateNotification()];
```

Comment:
Would lean towards putting the first two lines of each of these in an `beforeEach`, but not a blocker if you prefer as is. Similar for the 'Get' tests

Reply:
I have now added `beforeEach` where applicable.